### ISSUE TITLE
High memory usage while scanning for whitelists

### ISSUE THREAD

**Issue by gizellos** [revilheart/ESGST#396]
When I am scanning for whitelists, the memory usage is being bigger and bigger with each scanned page. I scanned cca 160 people and it took 6.8 GB of RAM. It was not like that before a time.

![2017-09-03_115029](https://user-images.githubusercontent.com/2397851/30001985-d25b5f64-909e-11e7-8955-62f7099254c9.png)

**Comment by revilheart** [comment]
I cannot reproduce this. Are you absolutely certain it's related to the checker? What were the options you had enabled? How many pages were scanned? How many people resulted in having whitelisted you?

**Comment by gizellos** [comment]
Yes, checker do this. I don't have any other problems with memory. Check this gif. It is quite fast. I used everything except the clearing cache option. No matter how many pages is scanned the consumption is higher with each the page.

![memory usage](https://user-images.githubusercontent.com/2397851/30027416-3a80699e-9181-11e7-84d0-f91f6f72352f.gif)

**Comment by revilheart** [comment]
The consumption in the GIF never reaches 6GB though, it appears to stay going up and down, always hovering below 2GB, which is how it's always been as far as I know. Are you absolutely certain it wasn't like this before?

**Comment by gizellos** [comment]
I didn't want to make the gif too long. If it were long, it would continue to grow until the end of scanning.

I am not absolutely sure it was not like that before but I believe it because I have never had problems like crashing chrome due to low memory. That's how I found it out. 

I would like to try a similar feature that load pages. I tried to look for a key and it scanned about 20 pages and the memory usage was low. But 20 is not many for a good test.
